### PR TITLE
Highlight by legend

### PR DESCRIPTION
**Pull request by ilgizar**
I propose my solution to the #6932. Without highlighting by the panel, only by the legend.

PS: bug #10545 is clearly manifested

### CODE REVIEW

File: public/app/plugins/panel/graph/graph.ts [hunk 1]
```diff
@@ -384,6 +386,57 @@ function graphDirective(timeSrv, popoverSrv, contextSrv) {
...
+            highlightWidth =
+              ('lineWidth' in sortedSeries[i].dashes
+                ? sortedSeries[i].dashes.lineWidth
+                : options.series.lines.lineWidth) + 1;
+            if (highlightWidth > 1) {
+              sortedSeries[i].dashes.lineWidth = highlightWidth;
+            }
+          }
+
+          if (options.series.bars.show || sortedSeries[i].bars.show) {
```
Comment (Ijin08) [inline public/app/plugins/panel/graph/graph.ts]: This is a nice feature. Great work.

But I don't think this effect on the bars is that useful. For me when I hover the legend the change to the bars are so small I can't se them. So for now I think we can remove highlight from the bars.

**Review by Ijin08 — CHANGES_REQUESTED** [review]

File: public/app/plugins/panel/graph/graph.ts (hunk 1 above)
Comment (ilgizar) [reply public/app/plugins/panel/graph/graph.ts]: It is easier to pull down than to build.

**Review by ilgizar — COMMENTED** [review]
Remove highlight from the bars.

**Review by Ijin08 — APPROVED** [review]

**Comment by daniellee** [comment]
@ilgizar Patrick has already reviewed the functionality of this feature (it works great!). 

I have question about the code. In graph.ts, in the highlightSeries function, there is a lot of duplication - setting highlight width for lines, dashes and points is very similar. Is it possible to extract the duplicate code to a function and change the `highlightSeries` function to look like this (or something similar):

```ts
      function highlightSeries(options) {
        if (ctrl.highlighting === undefined) {
          return;
        }

        for (let i = 0; i < sortedSeries.length; i++) {
          if (ctrl.highlighting !== sortedSeries[i].alias) {
            continue;
          }

          setHighlightWidth(options.series, sortedSeries[i], 'lines', 'lineWidth');
          setHighlightWidth(options.series, sortedSeries[i], 'dashes', 'lineWidth');
          setHighlightWidth(options.series, sortedSeries[i], 'points', 'radius');

          break;
        }
      }
```

Maybe the function could look something like this (this could be abstracted even more to remove the duplication with series[drawMode] and sortedSerie[drawMode]):

```ts
function setHighlightWidth(series, sortedSerie, drawMode, attribute) {
  let highlightWidth = 0;

  if (series[drawMode].show || sortedSerie[drawMode].show) {
    highlightWidth =
      (attribute in sortedSerie[drawMode] ? sortedSerie[drawMode][attribute] : series[drawMode][attribute]) + 1;
    if (highlightWidth > 1) {
      sortedSerie[drawMode][attribute] = highlightWidth;
    }
  }
}
```

**Comment by ilgizar** [comment]
@daniellee Excellent suggestion to improve the readability of the code and avoiding duplication.

**Comment by torkelo** [comment]
closing this due to inactivity

**Pull request closed without merging**